### PR TITLE
Fix to output 9 soil levels for RRFS

### PR DESCRIPTION
**Pull request by EricJames-NOAA**
Previously, UPP was defaulting to 4 soil levels.  These changes increase flexibility to check number of soil levels in FV3 netCDF files, and output appropriate number (currently no more than 9 levels).  

The change was tested in retrospective mode on Jet for RRFS_NA_3km domain.  

There is a change to the postxconfig file, which will be in a separate regional_workflow PR.

### CODE REVIEW

File: sorc/ncep_post.fd/WRFPOST.f [hunk 1]
```diff
@@ -444,6 +444,26 @@ PROGRAM WRFPOST
...
+          endif
+! read in LSM index and nsoil here
+          Status=nf90_get_att(ncid2d,nf90_global,'landsfcmdl', &
+                 iSF_SURFACE_PHYSICS)
+          if(Status/=0)then
+            print*,'landsfcmdl not found; assigning to 2'
+            iSF_SURFACE_PHYSICS=2 !set LSM physics to 2 for NOAH
+          endif
+          Status=nf90_get_att(ncid2d,nf90_global,'nsoil', &
+                 NSOIL)
```
Comment (guoqing-noaa) [inline sorc/ncep_post.fd/WRFPOST.f]: cosmetic suggestion, it looks like line 459 and 460 can be put into one line. Similar for Line 453,454.

File: sorc/ncep_post.fd/INITPOST_NETCDF.f [hunk 2]
```diff
@@ -1039,13 +1031,13 @@ SUBROUTINE INITPOST_NETCDF(ncid3d)
...
-        print*,'error opening ',fileNameFlux, ' Status = ', Status
-        print*,'skip reading of flux file' 
-      endif
+!! open flux file
+!      Status = nf90_open(trim(fileNameFlux),NF90_NOWRITE, ncid2d)
+! 
+!      if ( Status /= 0 ) then
+!        print*,'error opening ',fileNameFlux, ' Status = ', Status
+!        print*,'skip reading of flux file' 
+!      endif
```
Comment (guoqing-noaa) [inline sorc/ncep_post.fd/INITPOST_NETCDF.f:1040]: Just curious why to comment out flux file opening?

File: parm/fv3lam_rrfs.xml [hunk 3]
```diff
@@ -385,55 +385,22 @@
...
-
        <param>
        <shortname>TSOIL_ON_DEPTH_BEL_LAND_SFC</shortname>
        <pname>TSOIL</pname>
-       <scale_fact_fixed_sfc1>2 2 2 2</scale_fact_fixed_sfc1>
-       <level>0. 10. 40. 100.</level>
-       <scale_fact_fixed_sfc2>2 2 2 2</scale_fact_fixed_sfc2>
-       <level2>10. 40. 100. 200.</level2>
+       <level>0. 1. 4. 10. 30. 60. 100. 160. 300.</level>
+       <level2>0. 1. 4. 10. 30. 60. 100. 160. 300.</level2>
```
Comment (guoqing-noaa) [inline parm/fv3lam_rrfs.xml:392]: One question: do fv3lam_rrfs.xml and postxconfig-NT-fv3lam_rrfs.txt work for NOAA MP LSM?  I know EMC also runs a version of RRFS at WCOSS but they use NOAA LSM.

File: sorc/ncep_post.fd/WRFPOST.f (hunk 1 above)
Comment (EricJames-NOAA) [reply sorc/ncep_post.fd/WRFPOST.f]: Thanks, good idea.  I'll make that change.

File: sorc/ncep_post.fd/INITPOST_NETCDF.f (hunk 2 above)
Comment (EricJames-NOAA) [reply sorc/ncep_post.fd/INITPOST_NETCDF.f:1040]: Yeah, I moved the opening of this file to the parent code WRFPOST.f in order to get some attributes earlier on.  I could just delete these lines, but not sure what would happen if the code tried to open the file twice.

File: parm/fv3lam_rrfs.xml (hunk 3 above)
Comment (EricJames-NOAA) [reply parm/fv3lam_rrfs.xml:392]: I believe *fv3lam_rrfs* files are only for our GSL runs.  Maybe @hu5970 can confirm that...

**Pull request closed without merging**